core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -46,16 +46,34 @@ object ZkUtils {
   val ClusterIdPath = "/cluster/id"
   val BrokerIdsPath = "/brokers/ids"
   val BrokerTopicsPath = "/brokers/topics"
-  val ControllerPath = "/controller"
-  val ControllerEpochPath = "/controller_epoch"
   val ReassignPartitionsPath = "/admin/reassign_partitions"
   val DeleteTopicsPath = "/admin/delete_topics"
   val PreferredReplicaLeaderElectionPath = "/admin/preferred_replica_election"
   val BrokerSequenceIdPath = "/brokers/seqid"
-  val IsrChangeNotificationPath = "/isr_change_notification"
-  val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
 
+  // Important: it is necessary to add any new persistent Zookeeper path here
```

Comment:
/controller is actually not a persistent path. So, we can just say top level Zookeeper path.